internal/config.ts

```diff
@@ -0,0 +1,52 @@
+import { execute } from "./execute";
+import { Param, JSParamValues, ParamKinds } from "./parameters";
+import { RuntimeKind } from "./runtime";
+
+export type ParamKind<PSchema extends ParamKinds | Param> = PSchema extends Param
+  ? PSchema["kind"]
+  : PSchema;
+
+export type Params = Record<string, ParamKinds | Param>;
+
+export type ParamValues<TParams extends Params> = {
+  [PSlug in keyof TParams]: JSParamValues[ParamKind<TParams[PSlug]>];
```

Comment:
The `T<...>`convention just helps skim a type and pick out which types are generics and which aren't:

```suggestion
  [TSlug in keyof TParams]: JSParamValues[ParamKind<TParams[TSlug]>];
```